package-system/OpenSSL/FindOpenSSL.cmake.template

```diff
@@ -6,29 +6,65 @@
 #
 #
 
-set(TARGET_WITH_NAMESPACE "3rdParty::OpenSSL")
-if (TARGET $${TARGET_WITH_NAMESPACE})
+set(OPENSSL_O3DE_NAMESPACE "3rdParty::OpenSSL")
+if (TARGET $${OPENSSL_O3DE_NAMESPACE})
     return()
 endif()
 
+set(SSL_TARGETNAME "OpenSSL::SSL")
+set(CRYPTO_TARGETNAME "OpenSSL::Crypto")
+
+# we're trying to be a drop-in replacement for the FindOpenSSL.cmake that is shipped
+# with CMake itself, so we set the same variables with the same uppercase for compatibility
+# for questions about these variables, see https://cmake.org/cmake/help/latest/module/FindOpenSSL.html
+set(OPENSSL_FOUND True)
 set(OPENSSL_INCLUDE_DIR $${CMAKE_CURRENT_LIST_DIR}/OpenSSL/include)
-set(OPENSSL_LIBS_DIR $${CMAKE_CURRENT_LIST_DIR}/OpenSSL/$$<$$<CONFIG:debug>:debug/>lib)
+# c-only pakcages can be released containing only Release executables
```

Comment:
Typo. 
```
# C-only Packages
        ~~~~~~~
```

Reply:
Yes, I'm working on the upgrade for other platforms right now. I can keep this PR open and add any additional changes for other platform to it for review if that helps to save us some effort and time.